crates/wasm-smith/src/core.rs

```diff
@@ -1055,12 +1083,20 @@ impl Module {
 
     fn arbitrary_ref_type(&self, u: &mut Unstructured) -> Result<RefType> {
         if !self.config.reference_types_enabled {
-            return Ok(RefType::FUNCREF);
+            // Create a `RefType::FUNCREF` but with variable sharedness.
+            Ok(RefType {
+                nullable: true,
+                heap_type: HeapType::Abstract {
+                    shared: self.arbitrary_shared(u)?,
+                    ty: AbstractHeapType::Func,
+                },
+            })
```

Comment:
Given that shared-everything-threads now currently requires reference-types the changes here might be reasonable to undo